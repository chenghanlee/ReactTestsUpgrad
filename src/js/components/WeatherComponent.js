import React, { Component } from 'react';
import 'babel-polyfill';

import '../../css/tachyons.min.css';

const fetchJsonResponse = async url => {
    try {
        const response = await fetch(url);
        const responseJson = await response.json();
        return responseJson;
    } catch (e) {
        console.log('fetch url failed');
    }
};

export default class Weather extends Component {
    constructor(props) {
        super(props);
        this.state = {
            city: '',
            temperature: '',
        };
    }

    componentDidMount() {
        const weatherUrl = this.props.url;

        return fetchJsonResponse(weatherUrl).then(resp => {
            const city = resp.city;
            const temperature = resp.temp;

            this.setState({
                city: city,
                temperature: temperature,
            });
        });
    }

    render() {
        const background = 'http://mrmrs.github.io/photos/display.jpg';
        return (
            <article className="helvetica pb5">
                <header className="vh-100 bg-light-pink dt w-100">
                    <div
                        style={{ backgroundImage: `url(${background}) no-repeat center right background-size: cover` }}
                        className="dtc v-mid cover ph3 ph4-m ph5-l"
                    >
                        <h1 className="f2 f-subheadline-l measure lh-title fw9">
                            Weather at {this.state.city} is current {this.state.temperature}
                        </h1>
                        <h2 className="f6 fw6 black">A simple weather component</h2>
                    </div>
                </header>
            </article>
        );
    }
}
