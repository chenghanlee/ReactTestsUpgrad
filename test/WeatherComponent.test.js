import React from 'react';
import fetchMock from 'fetch-mock';
import Enzyme, { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import WeatherComponent from '../src/js/components/WeatherComponent';

Enzyme.configure({ adapter: new Adapter() });

fetchMock.get(`*`, JSON.stringify({ city: `Dallas`, temp: '90' }));

test(`Component mounts correctly while using async await`, async () => {
    const weather = shallow(
        <WeatherComponent url="https://api.darksky.net/forecast/0d2687a225ecee064c93466b5738f1fb/37.8267,-122.4233" />
    );

    // mounting the component
    await weather.instance().componentDidMount();

    // checks to see if the state is mounted correctly during componentDidMount
    expect(weather.state()).toHaveProperty('city', 'Dallas');
    expect(weather.state()).toHaveProperty('temperature', '90');
});

test('Component renders correctly', () => {
    const weather = shallow(<WeatherComponent />);
    expect(shallowToJson(weather)).toMatchSnapshot();
});
