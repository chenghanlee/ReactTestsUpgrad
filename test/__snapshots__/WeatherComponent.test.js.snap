// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<article
  className="helvetica pb5"
>
  <header
    className="vh-100 bg-light-pink dt w-100"
  >
    <div
      className="dtc v-mid cover ph3 ph4-m ph5-l"
      style={
        Object {
          "backgroundImage": "url(http://mrmrs.github.io/photos/display.jpg) no-repeat center right background-size: cover",
        }
      }
    >
      <h1
        className="f2 f-subheadline-l measure lh-title fw9"
      >
        Weather at 
         is current 
      </h1>
      <h2
        className="f6 fw6 black"
      >
        A simple weather component
      </h2>
    </div>
  </header>
</article>
`;
